libp2p/protocols/secure/secure.nim

```diff
@@ -52,8 +53,9 @@ method initStream*(s: SecureConn) =
   procCall Connection(s).initStream()
 
 method close*(s: SecureConn) {.async.} =
+  trace "closing secure conn", s, dir = s.dir
   if not(isNil(s.stream)):
-    await s.stream.close()
+    await s.stream.closeWithEOF()
```

Comment:
this looks wrong - just because we stop sending doesn't mean we've read everything there was to read? ie `closeWithEOF` should only be used when there is an in-protocol signal that all reading is finished, for example like in eth2 where you read one message and that's it

Reply:
Yeah, this was mostly an experiment to see if it synchronizes closes between the child and the parent streams. It doesn't seem like it made any difference so reverting back.